packages/pn-commons/src/components/CodeModal/CodeInput.tsx

```diff
@@ -116,6 +117,17 @@ const CodeInput = ({ initialValues, isReadOnly, hasError, onChange }: Props) =>
     }
   };
 
+  const pasteHandler = (event: ClipboardEvent<HTMLDivElement>, index: number) => {
+    const copiedCode = event.clipboardData.getData('text');
+    if (Number(copiedCode)) {
+      const values = copiedCode.split('');
+      values.map((element, i) => {
```

Comment:
Avoid to do this, because for each call to changeInputValue and focusInput, we re-render the component.
So call the setCurrentValues overriding all the values only once.
`
setCurrentValues(values);
`